smoke-test/src/test/java/org/opennms/smoketest/JmxConfigurationGeneratorIT.java

```diff
@@ -106,39 +110,79 @@ public void verifyCompMemberSelection() throws InterruptedException {
         Assert.assertEquals(7, find("<comp-member", jmxDatacollectionConfigContent));
     }
 
-    // switches to the embedded vaadin iframe
-    private void switchToVaadinFrame() {
-        // switchTo() by xpath is much faster than by ID
-        m_driver.switchTo().frame(findElementByXpath("/html/body/div/iframe"));
-    }
+    protected void configureJMXConnection(final boolean skipDefaultVM) throws Exception {
+        final long end = System.currentTimeMillis() + LOAD_TIMEOUT;
+
+        final WebDriverWait shortWait = new WebDriverWait(m_driver, 10);
 
-    private void updateConnection() {
-        updateElementValue("port", "18980");
+        final String skipDefaultVMxpath = "//span[@id='skipDefaultVM']/input";
+        final boolean selected = waitForElement(By.xpath(skipDefaultVMxpath)).isSelected();
+        LOG.debug("skipDefaultVM selected: {}", selected);
+        if (selected != skipDefaultVM) {
+            waitForElement(By.xpath(skipDefaultVMxpath)).click();
+        }
 
         // configure authentication
-        if (!findElementById("authenticate").isSelected()) {
-            findElementById("authenticate").findElement(By.tagName("input")).click();
+        final WebElement authenticateElement = waitForElement(By.id("authenticate"));
+        if (!authenticateElement.isSelected()) {
+            authenticateElement.findElement(By.tagName("input")).click();
         };
-        updateElementValue("authenticateUser", "admin");
-        updateElementValue("authenticatePassword", "admin");
+
+        boolean found = false;
+        do {
+            setVaadinValue("port", "18980");
+            setVaadinValue("authenticateUser", "admin");
+            setVaadinValue("authenticatePassword", "admin");
+
+            // go to next page
+            waitForElement(By.id("next")).click();
+
+            try {
```

Comment:
Can you add some comments to all of these lines to outline why you had to do all of these backflips? It might be difficult to understand for anyone coming into this code later.